tools/generator/src/Generator/SemanticVersion.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+
+// Inspired by https://gist.github.com/mjdescy/a805b5b4c49ed79fb240d3886815d5a2
+struct SemanticVersion: Equatable {
+    public let major: Int
+    public let minor: Int
+    public let patch: Int
+
+    public init?(version: String) {
+        let components = version.split(separator: ".")
+        guard components.count == 3 else { return nil }
```

Comment:
If the count is less than 3, we can assume the other parts are `0`. And if it's more than 3, do we care?

Reply:
I can move the logic that we currently have in `Platform.OSVersion` to here. I was relying on the current `OSVersion` logic to have already executed. Perhaps, I can cover this in the refactor that we discussed [here](https://github.com/buildbuddy-io/rules_xcodeproj/pull/856#discussion_r935855002).

Reply:
Done